drivers/sensor/bmp085/bmp085.c

```diff
@@ -0,0 +1,334 @@
+	k_msleep(26);
+#endif
+
+	ret = bmp085_read(dev, BMP085_REG_MSB, buf, sizeof(buf));
+	if (ret < 0) {
+		LOG_DBG("Failed to read MSB.");
+		return ret;
+	}
+
+	bmp085->raw_press = buf[1] | (buf[0] << 8);
+	bmp085->raw_press <<= 8;
+
+	ret = bmp085_read(dev, BMP085_REG_XLSB, &buf_xlsb, sizeof(buf_xlsb));
+	if (ret < 0) {
+		LOG_DBG("Failed to read MSB.");
+		return ret;
+	}
+
+	bmp085->raw_press |= buf_xlsb;
+	bmp085->raw_press >>= (8 - oversampling);
+
+	return ret;
+}
+
+static int bmp085_read_temp(const struct device *dev)
+{
+	struct bmp085_device_data *bmp085 = dev->data;
+	int ret = 0;
+
+	ret |= bmp085_read_raw_temp(dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	int32_t X1 = (bmp085->raw_temp - (int32_t)bmp085->bmpAC6) * ((int32_t)bmp085->bmpAC5) >> 15;
+	int32_t X2 = ((int32_t)bmp085->bmpMC << 11) / (X1 + (int32_t)bmp085->bmpMD);
+	int32_t B5 = X1 + X2;
+
+	bmp085->temp = (B5 + 8) >> 4;
+	LOG_INF("bmp085->temp: %d.%d°C", bmp085->temp / 10, bmp085->temp % 10);
+
+	return 0;
+}
+
+static int bmp085_read_press(const struct device *dev, uint8_t oversampling)
+{
+	struct bmp085_device_data *bmp085 = dev->data;
+	int ret = 0;
+
+	ret |= bmp085_read_raw_temp(dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret |= bmp085_read_raw_pres(dev, oversampling);
+	if (ret != 0) {
+		return ret;
+	}
+
+	int32_t B3, B5, B6, X1, X2, X3;
+	uint32_t B4, B7;
+
+	/* See datasheet, page 13
+	 * Calculation of pressure and temperature
+	 * Datasheet: https://mm.digikey.com/Volume0/opasdata/d220001/medias/docus/1085/BMP085.pdf
+	 */
+
+	X1 = (bmp085->raw_temp - (int32_t)bmp085->bmpAC6) * ((int32_t)bmp085->bmpAC5) >> 15;
+	X2 = ((int32_t)bmp085->bmpMC << 11) / (X1 + (int32_t)bmp085->bmpMD);
+	B5 = X1 + X2;
+
+	B6 = B5 - 4000;
+	X1 = ((int32_t)bmp085->bmpB2 * ((B6 * B6) >> 12)) >> 11;
+	X2 = ((int32_t)bmp085->bmpAC2 * B6) >> 11;
+	X3 = X1 + X2;
+	B3 = ((((int32_t)bmp085->bmpAC1 * 4 + X3) << oversampling) + 2) / 4;
+
+	X1 = ((int32_t)bmp085->bmpAC3 * B6) >> 13;
+	X2 = ((int32_t)bmp085->bmpB1 * ((B6 * B6) >> 12)) >> 16;
+	X3 = ((X1 + X2) + 2) >> 2;
+	B4 = ((uint32_t)bmp085->bmpAC4 * (uint32_t)(X3 + 32768)) >> 15;
+	B7 = ((uint32_t)bmp085->raw_press - B3) * (uint32_t)(50000UL >> oversampling);
+
+	if (B7 < 0x80000000) {
+		bmp085->press = (B7 * 2) / B4;
+	} else {
+		bmp085->press = (B7 / B4) * 2;
+	}
+	X1 = (bmp085->press >> 8) * (bmp085->press >> 8);
+	X1 = (X1 * 3038) >> 16;
+	X2 = (-7357 * bmp085->press) >> 16;
+
+	bmp085->press = bmp085->press + ((X1 + X2 + (int32_t)3791) >> 4);
+	LOG_INF("bmp085->press: %dPa", bmp085->press);
+
+	return 0;
+}
+
+static int bmp085_read_cal_coef(const struct device *dev)
+{
+	struct bmp085_device_data *bmp085 = dev->data;
+	int ret = 0;
+	uint8_t buf[22];
+
+	ret = bmp085_read(dev, BMP085_REG_CAL_COEF, buf, sizeof(buf));
+	if (ret < 0) {
+		LOG_DBG("Failed to read calibration coefficients.");
+		return ret;
+	}
+
+	bmp085->bmpAC1 = buf[1] | (buf[0] << 8);
+	LOG_INF("bmpAC1: %d/0x%X", bmp085->bmpAC1, bmp085->bmpAC1);
```

Comment:
LOG_DBG would be more appropriate IMO (or maybe no log at all?)

If the logs stay, you may want to look at using the same format string everywhere to save a few bytes in flash (`"%s: %d/0x%X"`)